packages/palette_generator/example/lib/main.dart

```diff
@@ -90,20 +92,21 @@ class _ImageColorsState extends State<ImageColors> {
 
   // Called when the user starts to drag
   void _onPanDown(DragDownDetails details) {
-    final RenderBox box = imageKey.currentContext.findRenderObject();
+    final RenderBox box =
+        imageKey.currentContext!.findRenderObject() as RenderBox;
     final Offset localPosition = box.globalToLocal(details.globalPosition);
     setState(() {
       startDrag = localPosition;
-      currentDrag = startDrag;
-      dragRegion = Rect.fromPoints(startDrag, currentDrag);
+      currentDrag = startDrag!;
+      dragRegion = Rect.fromPoints(startDrag!, currentDrag!);
```

Comment:
Why are all of these `!`s safe? If they are required to be set, then they should be declared non-null.

Reply:
these are set as `null` in `_onPanCancel` and in `_onPanCancel` methods  https://github.com/flutter/packages/blob/48a2148811a09b6354e8b02551e34d156ed56916/packages/palette_generator/example/lib/main.dart#L112 and if I understand correctly `_onPanUpdate` is always called before these methods so these variables should not be null when accessed